### ISSUE TITLE
Displaying Duration as hh:mm:ss

### ISSUE THREAD

**Issue by paul-lrr** [moment/luxon#214]
I have been getting some strange results when formatting a Duration as `hh:mm:ss` (for a timer). When the Duration contains both seconds and milliseconds, the seconds unit can exceed 59. It seems to be adding the milliseconds to the total _after_ the formatting calculation has been done. 
eg. `Duration.fromObject({seconds:51,milliseconds:30000}).toFormat('hh:mm:ss')`
outputs: `"00:00:81"` where it should be `"00:01:21"`

In a related issue, even if the Duration is only using milliseconds, the rounding also seems to happen _after_ the formatting calculation.
eg. `Duration.fromObject({milliseconds:59700}).toFormat('hh:mm:ss')`
outputs: `"00:00:60"` where it should be `"00:01:00"`
this also means that a timer incrementing every second by adding 1000ms will go from `"00:00:60"` to `"00:01:01"`

Is there a better way I should be using to to display a Duration in a easily readable hh:mm:ss format? Alternatively, is there a way to round a Duration to the desired unit? If I  just convert a Duration to seconds using `.shiftTo('seconds')` or Diff two DateTime objects using seconds as the unit, it results in a fractional second, which still has the above rounding issue.
eg. `Duration.fromObject({milliseconds:60789}).shiftTo('seconds').toObject()`
outputs: `{seconds: 60.789}`

So far the only solution I have found is to round the values I am going to use in the Duration _before_ I make it and to apply `.startOf('seccond')` to DateTimes but that seems like it should be an unnecessary step.

Cheers,
Paul

**Comment by icambron** [comment]
The first was was fixed in 0.5.5. Or at least, it works now and it looks a lot like a `shiftTo` issue I just fixed:

```js
Duration.fromObject({seconds:51,milliseconds:30000}).toFormat('hh:mm:ss') //=> '00:01:21'
```

The second one is still there:

```js
Duration.fromObject({milliseconds:59700}).toFormat('hh:mm:ss') //=> '00:00:60'

// more to the point
Duration.fromObject({seconds:59.7}).toFormat('hh:mm:ss') // => '00:00:60'
```

Instead of trying to do the unit cascade before rounding (which is complicated - the rounding happens as part of converting the number to a string), I think I just want to truncate to 59 there. We've handled analogous situations that way.

Open to feedback on that

**Comment by icambron** [comment]
I have a fix for this in 9ab298b. Will release it shortly.

**Comment by prkagrawal** [comment]
Thanks, this issue helped me.
